### PR TITLE
Added 7 days trade hold exception and Team Fortress 2 ID's.

### PR DESCRIPTION
**Pull request by mactepp**
I got this error and don't know if it is counting the 7 days or resetting every time I log in. By the way, nice work here, mate.

### CODE REVIEW

File: steampy/client.py [hunk 1]
```diff
@@ -181,7 +184,11 @@ def accept_trade_offer(self, trade_offer_id: str) -> dict:
     def _fetch_trade_partner_id(self, trade_offer_id: str) -> dict:
```
Comment (bukson) [inline steampy/client.py]: You can change type to def accept_trade_offer(self, trade_offer_id: str) -> str: 
I spotted that bug when i was reviewing your pull request

File: steampy/client.py [hunk 2]
```diff
@@ -181,7 +184,11 @@ def accept_trade_offer(self, trade_offer_id: str) -> dict:
...
         url = self._get_trade_offer_url(trade_offer_id)
         offer_response_text = self._session.get(url).text
-        return text_between(offer_response_text, "var g_ulTradePartnerSteamID = '", "';")
+        if "You have logged in from a new device. In order to protect the items" in offer_response_text:
+            trade_hold = False
```
Comment (bukson) [inline steampy/client.py]: Bad practise, dont return Bool when it should return String. You can add Exception SevenDaysHoldException and raise it there.

File: steampy/client.py [hunk 3]
```diff
@@ -165,6 +165,9 @@ def get_trade_offer(self, trade_offer_id: str, merge: bool = True) -> dict:
     @login_required
     def accept_trade_offer(self, trade_offer_id: str) -> dict:
         partner = self._fetch_trade_partner_id(trade_offer_id)
+        if partner == False:
```
Comment (bukson) [inline steampy/client.py]: Delete this, because it will raise exception. You can also add to documantation that accept_trade_offer raises exception.

Comment (bukson) [inline steampy/client.py]: Thank you for your input :) I forgot about TF2

**Comment by mactepp** [comment]
I just changed some thing you asked for. Sorry about bad practices, I'm new to programming (and was I little bit sleepy hehehe). Take a look and tell me if it is good now

File: steampy/client.py [hunk 4]
```diff
@@ -164,10 +164,10 @@ def get_trade_offer(self, trade_offer_id: str, merge: bool = True) -> dict:
...
     @login_required
     def accept_trade_offer(self, trade_offer_id: str) -> dict:
-        partner = self._fetch_trade_partner_id(trade_offer_id)
-        if partner == False:
-            print("Account on hold, cannot accept trades")
-            return None
+        try:
+            partner = self._fetch_trade_partner_id(trade_offer_id)
+        except SevenDaysHoldException:
+            return
```
Comment (bukson) [inline steampy/client.py]: Just self._fetch_trade_partner_id(trade_offer_id)
Exception should be caught outside accept_trade_offer method

File: steampy/client.py [hunk 5]
```diff
@@ -181,12 +181,11 @@ def accept_trade_offer(self, trade_offer_id: str) -> dict:
...
         return response
 
-    def _fetch_trade_partner_id(self, trade_offer_id: str) -> dict:
+    def _fetch_trade_partner_id(self, trade_offer_id: str) -> str:
         url = self._get_trade_offer_url(trade_offer_id)
         offer_response_text = self._session.get(url).text
         if "You have logged in from a new device. In order to protect the items" in offer_response_text:
-            trade_hold = False
-            return trade_hold
+            raise SevenDaysHoldException
```
Comment (bukson) [inline steampy/client.py]: can be SevenDaysHoldException('Account has logged in from  a new device and can't trade for 7 days') for more information in log

File: steampy/client.py [hunk 6]
```diff
@@ -181,12 +181,11 @@ def accept_trade_offer(self, trade_offer_id: str) -> dict:
             return self._confirm_transaction(trade_offer_id)
         return response
 
-    def _fetch_trade_partner_id(self, trade_offer_id: str) -> dict:
+    def _fetch_trade_partner_id(self, trade_offer_id: str) -> str:
         url = self._get_trade_offer_url(trade_offer_id)
         offer_response_text = self._session.get(url).text
         if "You have logged in from a new device. In order to protect the items" in offer_response_text:
```
Comment (bukson) [inline steampy/client.py]: I use single quote in whole project

'You have logged in from a new device. In order to protect the items'

File: steampy/client.py [hunk 7]
```diff
@@ -254,3 +253,7 @@ def fetch_price(self, item_hash_name: str, game: GameOptions, currency: str = Cu
                   'appid': game.app_id,
                   'market_hash_name': item_hash_name}
         return self._session.get(url, params=params).json()
+
+
+class SevenDaysHoldException(Exception):
+    """Account has logged in a new device and can't trade for 7 days"""
```
Comment (bukson) [inline steampy/client.py]: """Account has logged in from a new device and can't trade for 7 days"""

You are sure that every time user log in he won't have this exception.
Maybe steam thinks that every bot session is from new device?

File: steampy/client.py [hunk 8]
```diff
@@ -178,10 +178,13 @@ def accept_trade_offer(self, trade_offer_id: str) -> dict:
             return self._confirm_transaction(trade_offer_id)
         return response
 
-    def _fetch_trade_partner_id(self, trade_offer_id: str) -> dict:
+    def _fetch_trade_partner_id(self, trade_offer_id: str) -> str:
         url = self._get_trade_offer_url(trade_offer_id)
         offer_response_text = self._session.get(url).text
-        return text_between(offer_response_text, "var g_ulTradePartnerSteamID = '", "';")
+        if 'You have logged in from a new device. In order to protect the items' in offer_response_text:
+            raise SevenDaysHoldException('Account has logged in a new device and can\'t trade for 7 days')
```
Comment (bukson) [inline steampy/client.py]: can't, no can\'t

File: steampy/client.py [hunk 9]
```diff
@@ -178,10 +178,13 @@ def accept_trade_offer(self, trade_offer_id: str) -> dict:
...
+        else:
```
Comment (bukson) [inline steampy/client.py]: no need for else bacsue if in 184 line will raise exception

**Comment by bukson** [comment]
Please, after fixing that last issue, test if it works and maybe check about 7 days hold on bot.

File: steampy/client.py (hunk 8 above)
Comment (mactepp) [inline steampy/client.py]: If using single quote, the apostrophe needs to be escapped

Comment (bukson) [inline steampy/client.py]: Ofc, I forgot that :)

**Comment by mactepp** [comment]
It seems to be working, but I will keep testing. This error is no a countdown so I will have to wait until next week to see what happens. I will search for fixes anyway.

**Comment by bukson** [comment]
Ok merged i will deploy it on pip soon

**Comment by bukson** [comment]
Please inform me further with this issue

**Comment by mactepp** [comment]
Good news. I did some researches and people said that this error goes away when the authenticator gets 7 days old. I logged in the authenticator last week and today my bot just accepted the trade with no errors. By the way, I will test the exception because something strange happened, the bot ran without errors, but haven't accepted the trade. I don't know what is going on, but I will test it with another account. I think that will be good to put this note in the README.

**Comment by bukson** [comment]
Ok thank you very much, but how it is that authenticator have to be 14 days old to trade with others and you have 7 days hold time?

**Pull request merged**